### ISSUE TITLE
Images from the previous project still shown after opening a new project

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#3388]
## Describe the bug

When opening a project, sometimes the sprites have the colors of another sprite (in the scene editor).

Editing the object by double-clicking and hitting cancel seems to fix the rending problem and colors are drawn correctly.

## Videos

https://user-images.githubusercontent.com/8879811/146864769-47d93aaa-cdf0-4510-bf0d-3d693b8c19e9.mp4

## To Reproduce

I tried to recreate this but I can't find the exact trigger.  I know we need that information to solve the problem, but I wanted to document the problem so others can search for the specific conditions to repeat this.

## Other details
* Windows 10
* GDevelop 5.0.124

**Comment by AlexandreSi** [comment]
Hi @tristanbob!
Thanks for reporting this bug, it is weird!

Can you maybe provide a part of your project so that I can try reproduce it? You can send it to me privately as it is the one you use for your tutorials.

**Comment by tristanbob** [comment]
Yes I will try to get it to you today.  After chatting with @D8H we think it happens when you have two projects with the same object names.

**Comment by Midhil457** [comment]
I think it's the image that is being mixed and not the color. As in the given example project, the images are solid colors

**Comment by tristanbob** [comment]
Ok, this is SUPER simple to reproduce, but it takes two projects.  

1) Create a new scene and add a new sprite object.
2) Paint the object a unique color and give it a unique object name (but leave the default filename "NewObject")
3) Save and close the project.
4) Repeat steps 1-3 for a second project
5) Open first project that was created with the same steps
6) The Scene Editor will use the image from the previously opened project, at least until the object properties are opened, which causes the correct image to be used.

https://user-images.githubusercontent.com/8879811/148112992-38c3ceb5-9857-43cb-a3f5-bb44dc6c5fcd.mp4

## Example projects
[GDevelop projects.zip](https://github.com/4ian/GDevelop/files/7809947/GDevelop.projects.zip)

**Comment by Bouh** [comment]
- The tabs aren't updated until a confirmation button or a Ok button is clicked.
- Tabs aren't updated also when we switch from tab A to B and vice-versa.
- Also there is no watcher on resources so if a resource change of color outside of GDevelop the app don't will update it.

On this last point,[ I started something a long time ago,](https://github.com/4ian/GDevelop/pull/1724) but I dropped the PR due to the complexity.
If @AlexandreSi wants to take back the idea I can talk to you about my PR in a call.
